dlp/inspectStringCustomOmitOverlap.js

```diff
@@ -0,0 +1,111 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+// sample-metadata:
+//  title: Inspect a string for sensitive data, omitting custom matches.
+//  usage: node inspectStringCustomOmitOverlap.js my-project string
+function main(projectId, string) {
+  // [START dlp_inspect_string_custom_omit_overlap]
+  // Imports the Google Cloud Data Loss Prevention library
+  const DLP = require('@google-cloud/dlp');
+
+  // Instantiates a client
+  const dlp = new DLP.DlpServiceClient();
+
+  // The project ID to run the API call under
+  // const projectId = 'my-project';
+
+  // The string to inspect
+  // const string = 'Name: Jane Doe. Name: Larry Page.';
+
+  async function inspectStringCustomOmitOverlap() {
+    // Construct item to inspect
+    const item = {value: string};
+
+    // Construct a custom regex detector for names
+    const customInfoTypes = [
+      {
+        infoType: {name: 'VIP_DETECTOR'},
+        regex: {pattern: 'Larry Page|Sergey Brin'},
+        exclusionType:
+          DLP.protos.google.privacy.dlp.v2.CustomInfoType.ExclusionType
+            .EXCLUSION_TYPE_EXCLUDE,
+      },
+    ];
+
+    // Construct a exclusion rule
+    const exclusionRule = {
+      excludeInfoTypes: {
+        infoTypes: [{name: 'VIP_DETECTOR'}],
```

Comment:
Please use customInfoTypes.infoType.name instead?

Reply:
Will have to make it like `customInfoTypes[0].infoType` as customInfoTypes is a list. Means here I will be hardcoding the index. 2nd thing would be to do like `customInfoTypes.map(cInfoType => cInfoType.infoType)`. 2nd method i though would be confusing as a guide so thought to keep the string hardcoded. Which one do you think would be more appropriate?

Reply:
Ok. Let's keep it as is. I was comparing with Java where the code is a bit different.